src/core/qgsvectorlayer.h

```diff
@@ -1992,6 +2018,22 @@ class CORE_EXPORT QgsVectorLayer : public QgsMapLayer, public QgsExpressionConte
      */
     bool isEditCommandActive() const { return mEditCommandActive; }
 
+    /**
+     * The geometry options applied to this layer contain information about
+     * how geomtries should be preprocessed when added to or edited on this layer.
+     *
+     * \since QGIS 3.4
+     */
+    GeometryOptions geometryOptions() const;
```

Comment:
What do you think about returning a reference here instead?

Reply:
Is there a possibility to send a change signal with this approach?

Reply:
I'll likely need them in this case. And I'd even see that as a good reason to refrain from this approach in general ;)